internal/codespaces/api/api.go

```diff
@@ -686,3 +689,19 @@ func (a *API) do(ctx context.Context, req *http.Request, spanName string) (*http
 func (a *API) setHeaders(req *http.Request) {
 	req.Header.Set("Accept", "application/vnd.github.v3+json")
 }
+
+// withRetry takes a generic function that sends an http request and retries
+// only when the returned response has a >=500 status code.
+func (a *API) withRetry(f func() (*http.Response, error)) (resp *http.Response, err error) {
+	for i := time.Duration(0); i < 5; i++ {
```

Comment:
It feels odd that `i` is a `Duration`. Can't `i` just be a normal int and have the Duration-conversion bits later on where it's necessary?

Reply:
Thought this would be a little cleaner in the math below, but I'm happy either way, I'll move it over 👍🏼